src/main/java/duke/Task.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Task class represents a task with a description, status, and type.
+ */
+public class Task {
+    // inspired by partial solution on website
+    protected String description;
+    protected boolean isDone;
+    protected TaskType taskType;
+
+    /**
+     * Constructs a Task object with the specified description and task type.
+     *
+     * @param description The description of the task.
+     * @param taskType    The type of the task (ToDo, Deadline, or Event).
+     */
+    public Task(String description, TaskType taskType) {
+        this.description = description;
+        this.isDone = false;
+        this.taskType = taskType;
+    }
+
+    /**
+     * Returns an icon representing the status of the task.
+     *
+     * @return "X" if the task is done, " " (empty) if the task is not done.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as undone.
+     */
+    public void markAsUndone() {
```

Comment:
Not really an issue, but consider naming as `UnmarkAsDone` instead, to match with naming in the rest of the project.